bootstrap.md

```diff
@@ -0,0 +1,136 @@
+
+## What is bootstrapping in Filecoin?
+
+When a Filecoin node first comes online, it must find the latest head for the chain and then initiate the syncing routine in order to catch up to the chain state and maintain it thereafter as new blocks come in. This is covered in the [sync spec](./sync.md). Bootstrapping is the process through which a node finds the chain's latest head in the first place (thereby enabling it to sync), through other peers in the network.
+
+At its simplest, bootstrapping can be thought of as secure network joining. It must enable a new node to get the latest head/chain that the majority of the network power is mining off and must be both:
+
+- Robust — able to do so in adversarial networks (so long as honest nodes make up the majority of the network).
+
+- Consistent — able to reliably get the canonical chain (even given poor latency or state loss in our initial peer set).
+
+The specific bootstrapping heuristic could be left up to the node implementation, but we provide a simple one here as a candidate construction. This spec is divided as follows:
+
+- Initial Connection — ensuring the node connects to an honest set of initial peers.
+- Peer Set Expansion — ensuring that these peers can serve up the network's canonical head reliably in spite of their individual latencies or state loss.
+- Chain Selection — choosing the most likely canonical chain thereafter.
+
+We leave trustless initial peer selection as future work.
+
+## Initial Connection
+
+Initial peer selection is key to avoiding eclipse attacks. If the nodes you initially connect to are malicious, then they can simply serve you malicious other peers to all of your requests, and effectively prevent you from ever discovering the real network. Filecoin assumes 51% of rational or honest miners. Given no prior information, as is the case when bootstrapping, distinguishing a rational node from an adversarial one would require sampling a majority of the network.
+
+This is obviously impractical, so we choose to start the bootstrap sequence with nodes by connecting with an inital set of trusted peers. These can be nodes run by well-known entities (the Filecoin foundation, large exchanges, good actors in the community, etc) or manually specified by the node operator in the config file. We leave trustless (or fully decentralized) bootstrapping for future work.
+
+A strawman bootstrapping sequence would have nodes use these initial peers to sync up to the canonical chain. However, new connection requests might effecitvely DoS these peers, making the Filecoin network unavailable to new nodes. Therefore, Filecoin bootstrapping seeks to "load-balance" initial requests to ensure the network remains widely available.
+
+We connect to each of these initial peers using the `hello handshake` (see in [network protocols](./network-protocols.md#hello-handshake)), getting back
```

Comment:
minor clarification: we don't connect to them with the hello handshake, we connect and then handshake.